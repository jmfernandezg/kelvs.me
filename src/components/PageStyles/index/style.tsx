import styled from "styled-components";

export const HeroWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: calc(100vh - 7rem);
  margin-top: 3em;
  ${({ theme }) => theme.desktop`
    justify-content: center;
    margin-bottom: 0;
    margin-top: 0;
    min-height: calc(65vh - 7rem);
    `}
`;

// TODO: put a div in the hero that can be tilted like in react spring
export const HeroOne = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;
  font-family: "Caveat", cursive;
  h1 {
    display: inline-block;
    overflow: visible;
    width: 100%;
    flex: 0 auto;
    font-size: 4em;
    font-weight: 400;
    text-align: center;
    ${({ theme }) => theme.thone`font-size: 2em;`}
  }
  span {
    font-weight: 700;
  }
`;

export const HeroTwo = styled.div`
  h1 {
    position: relative;
    font-size: 3.5em;
    font-weight: 400;
    text-align: center;
    line-height: 58px;
    font-family: "Lato", sans-serif;
    ${({ theme }) => theme.thone`font-size: 2em;`}
    ${({ theme }) => theme.phablet`
      font-size: 1.5em;
      line-height: 35px;
    `}
  }
  span {
    padding-right: 5px;
    padding-left: 5px;
    background: linear-gradient(
      transparent 10%,
      #495867 10%,
      #495867 100%,
      transparent 90%
    );
    color: #fff;
    font-weight: 300;
    text-align: justify;
  }
`;

export const CallToAction = styled.div`
  margin: 4em 0;
  font-size: 15px;
  ${({ theme }) => theme.tablet`margin: 2em 0 0 0;`}
  a {
    padding: 0 20px;
    height: 25px;
  }
  a::before,
  a::after {
    position: absolute;
    width: 25px;
    height: 1px;
    background: ${props => props.theme.text.dark};
    content: "";
    opacity: 0.2;
    transition: all 0.3s;
    pointer-events: none;
  }
  .dark-mode & a::before,
  .dark-mode & a::after {
    background: ${props => props.theme.text.light};
  }
  a:hover {
    opacity: 0.9;
  }
  a::before {
    top: 0;
    left: 0;
    transform: rotate(90deg);
    transform-origin: 0 0;
  }
  a::after {
    right: 0;
    bottom: 0;
    transform: rotate(90deg);
    transform-origin: 100% 0;
  }
  a:hover::before,
  a:hover::after,
  a:focus::before,
  a:focus::after {
    opacity: 1;
  }
  a:hover::before,
  a:focus::before {
    left: 50%;
    transform: rotate(0deg) translateX(-50%);
  }

  a:hover::after,
  a:focus::after {
    right: 50%;
    transform: rotate(0deg) translateX(50%);
  }
`;

export const Contact = styled.div`
  width: 20%;
  display: flex;
  justify-content: space-around;
  svg {
    height: 1.5em;
    width: 100%;
    fill: ${props => props.theme.text.dark};
  }
  .dark-mode & svg {
    fill: ${props => props.theme.text.light};
  }
  ${({ theme }) => theme.bigTablet`
    margin: 2em 0;
    svg {
      height: 1.1em;
    }
  `}
  ${({ theme }) => theme.thone`
    width: 40%;
  `}
`;

export const FeaturedProjects = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
`;

export const AppBox = styled.div`
  background: #fff 0 0 no-repeat padding-box;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16);
  width: 32%;
  padding: 0.75em;
  margin-bottom: 1em;
  position: relative;
  transition: transform 0.2s;
  min-height: 290px;
  &:hover {
    transform: scale(1.025);
  }
  color: ${props => props.theme.text.dark};
  img {
    border: 1px solid #000;
    width: 100%;
    height: auto;
  }
  span {
    margin-right: 3px;
    background-color: #ff4e4e;
    color: #fff;
    padding: 0 3px 1px 3px;
    line-height: normal;
    font-size: 12px;
    font-weight: 300;
    letter-spacing: 1px;
    border-radius: 3px;
  }
  p {
    margin-bottom: 0;
  }
  ${({ theme }) => theme.tablet`
    width: 48%;
    p {
      font-size: 0.9rem;
    }
  `}
  ${({ theme }) => theme.thone`width: 100%;`}
`;

export const H1 = styled.h1`
  margin: 2.5em 0;
  ${({ theme }) => theme.tablet`
    margin: 2em 0;
    font-size: 1.125rem;
  `}
  font-weight: 700;
`;
