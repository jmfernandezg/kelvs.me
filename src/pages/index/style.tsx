import styled from "styled-components";

export const HeroWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  min-height: 40vh;
  ${({ theme }) => theme.tablet`justify-content: center;`}
`;

// TODO: put a div in the hero that can be tilted like in react spring
export const HeroOne = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;
  font-family: "Caveat", cursive;
  h1 {
    display: inline-block;
    overflow: visible;
    width: 100%;
    flex: 0 auto;
    font-size: 4em;
    font-weight: 400;
    text-align: center;
    ${({ theme }) => theme.thone`font-size: 2em;`}
  }
  span {
    font-weight: 700;
  }
`;

export const HeroTwo = styled.div`
  h1 {
    position: relative;
    font-size: 3.5em;
    font-weight: 400;
    text-align: center;
    line-height: 58px;
    font-family: "Lato", sans-serif;
    ${({ theme }) => theme.thone`font-size: 2em;`}
    ${({ theme }) => theme.phablet`
      font-size: 1.5em;
      line-height: 35px;
    `}
  }
  span {
    padding-right: 5px;
    padding-left: 5px;
    background: linear-gradient(
      transparent 10%,
      #495867 10%,
      #495867 100%,
      transparent 90%
    );
    color: #fff;
    font-weight: 300;
    text-align: justify;
  }
`;

export const CallToAction = styled.div`
  margin: 2em 0 4em 0;
  font-size: 15px;
  ${({ theme }) => theme.tablet`margin: 2em 0 4em 0;`}
  a {
    padding: 0 20px;
    height: 25px;
  }
  a::before,
  a::after {
    position: absolute;
    width: 25px;
    height: 1px;
    background: ${props => props.theme.text.dark};
    content: "";
    opacity: 0.2;
    transition: all 0.3s;
    pointer-events: none;
  }
  .dark-mode & a::before,
  .dark-mode & a::after {
    background: ${props => props.theme.text.light};
  }
  a:hover {
    opacity: 0.9;
  }
  a::before {
    top: 0;
    left: 0;
    transform: rotate(90deg);
    transform-origin: 0 0;
  }
  a::after {
    right: 0;
    bottom: 0;
    transform: rotate(90deg);
    transform-origin: 100% 0;
  }
  a:hover::before,
  a:hover::after,
  a:focus::before,
  a:focus::after {
    opacity: 1;
  }
  a:hover::before,
  a:focus::before {
    left: 50%;
    transform: rotate(0deg) translateX(-50%);
  }

  a:hover::after,
  a:focus::after {
    right: 50%;
    transform: rotate(0deg) translateX(50%);
  }
`;

export const Contact = styled.div`
  width: 20%;
  display: flex;
  justify-content: space-around;
  svg {
    height: 1.5em;
    width: 100%;
    fill: ${props => props.theme.text.dark};
  }
  .dark-mode & svg {
    fill: ${props => props.theme.text.light};
  }
  ${({ theme }) => theme.tablet`
    width: 40%;
    svg {
      height: 1.1em;
    }
  `}
`;
