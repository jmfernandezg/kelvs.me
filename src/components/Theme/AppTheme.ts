import { hex2rgba } from "../../helpers/index";
import media from "~/helpers/media";

const AppTheme = {
  bg: {
    light: "#fffffe",
    dark: "#232946",
  },
  text: {
    light: "#fffffe",
    dark: "#232946",
  },
  hover: {
    light: hex2rgba("#007aff", 0.1),
    dark: hex2rgba("##fffffe", 0.16),
  },
  font: {
    body:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol'",
    monospace: "SFMono-Regular,Consolas,Liberation Mono,Menlo,monospace",
  },
  fontSize: {
    xs: "0.75rem",
    sm: "0.875rem",
    base: "1rem",
    lg: "1.125rem",
    xl: "1.25rem",
    xxl: "1.5rem",
    x3l: "1.875rem",
  },
  fontWeight: {
    hairline: "100",
    thin: "200",
    light: "300",
    normal: "400",
    medium: "500",
    semibold: "600",
    bold: "700",
    extrabold: "800",
    black: "900",
  },
  transition: "all 0.25s cubic-bezier(0.645, 0.045, 0.355, 1)",
  ...media,
};

// https://spectrum.chat/styled-components/general/i-cant-use-my-theme-in-createglobalstyle-function-styled-components-v4-react-v16-6-3~0978b404-ab71-45c9-8f75-0862abde4eb5
export type ThemeType = typeof AppTheme;

export default AppTheme;
