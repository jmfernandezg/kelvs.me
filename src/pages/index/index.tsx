/* eslint-disable react/jsx-one-expression-per-line */
import Link from "next/link";
import Layout from "~/components/Layout";
import GmailSVG from "~/components/Icons/Gmail";
import GithubSVG from "~/components/Icons/Github";
import LinkedinSVG from "~/components/Icons/LinkedIn";
import { Container, HorizontalBar } from "~/components/PageStyles/common/style";
import {
  HeroWrapper,
  HeroOne,
  HeroTwo,
  CallToAction,
  Contact,
} from "~/components/PageStyles/index/style";

// https://www.justinewargnier.com/
// https://simpleicons.org/
// TODO: animate full stack dev text switch using react springs
export default function Home() {
  return (
    <Layout>
      <Container>
        <HeroWrapper>
          <HeroOne>
            <h1>
              Hello,&nbsp;
              <span>my name is Kelvin.</span>
            </h1>
            <HorizontalBar />
          </HeroOne>
          <HeroTwo>
            <h1>
              I&apos;m a <span>full&#8209;stack developer</span>
              &nbsp;and an open source enthusiast.
            </h1>
          </HeroTwo>
        </HeroWrapper>
        <CallToAction>
          <Link href="/about">
            <a>Learn more about me</a>
          </Link>
        </CallToAction>
        <Contact>
          <a href="https://github.com/kelvinabella">
            <GithubSVG />
          </a>
          <a href="https://www.linkedin.com/in/kelvinabella">
            <LinkedinSVG />
          </a>
          <a href="mailto:abellakelvin.ka@gmail.com">
            <GmailSVG />
          </a>
        </Contact>
      </Container>
    </Layout>
  );
}
