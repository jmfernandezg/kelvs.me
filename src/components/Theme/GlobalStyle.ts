import { createGlobalStyle } from "styled-components";
import { normalize } from "styled-normalize";
import { ThemeType } from "./AppTheme";

const GlobalStyle = createGlobalStyle<{ theme: ThemeType }>`
  ${normalize}

  *, *:before, *:after {
    box-sizing: border-box;
  }
  html {
    width: 100%;
    box-sizing: border-box;
    scroll-behavior: smooth;
  }
  body {
    margin: 0;
    width: 100%;
    min-height: 100%;
    overflow-x: hidden;
    -moz-osx-font-smoothing: grayscale;
    -webkit-font-smoothing: antialiased;
    background-color: ${props => props.theme.bg.light};
    color: ${props => props.theme.text.dark};
    line-height: 1.3;
    font-family: ${props => props.theme.font.body};
    font-size: ${props => props.theme.fontSize.base};
    ${({ theme }) =>
      theme.phablet`font-size: ${(props: any) => props.theme.fontSize.lg};`}

    &.dark-mode {
      background: ${props => props.theme.bg.dark};
      color: ${props => props.theme.text.light};
    }
  }

  a {
    display: inline-block;
    text-decoration: none;
    text-decoration-skip-ink: auto;
    color: inherit;
    position: relative;
    transition: all 0.25s cubic-bezier(0.645, 0.045, 0.355, 1);
    cursor: pointer;
  }

  ul {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 600;
    margin: 0;
  }

`;

export default GlobalStyle;
