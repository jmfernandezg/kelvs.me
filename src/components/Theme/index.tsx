import { ThemeProvider } from "styled-components";
import React from "react";
import GlobalStyle from "./GlobalStyle";
import AppTheme from "./AppTheme";

type Props = {
  children: React.ReactNode;
};

// https://www.typescriptlang.org/docs/handbook/jsx.html#the-jsx-result-type
export default function Theme({ children }: Props) {
  return (
    <ThemeProvider theme={AppTheme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
}
