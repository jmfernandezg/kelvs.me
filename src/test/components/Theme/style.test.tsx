/* eslint-disable react/no-children-prop */
import React from "react";
import { render, screen } from "~/test/utils/index";
import "jest-styled-components";
import GlobalStyle from "~/components/Theme/GlobalStyle";
import Theme from "~/components/Theme";

describe("Theme styles", () => {
  it("renders GlobalStyle correctly", () => {
    const { container } = render(<GlobalStyle />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders Theme correctly", () => {
    const { container } = render(<Theme children={<div>render div</div>} />);
    expect(container.firstChild).toMatchSnapshot();

    // render div must be found in the document
    const children = screen.getByText("render div");
    expect(children).toBeInTheDocument();
  });
});
