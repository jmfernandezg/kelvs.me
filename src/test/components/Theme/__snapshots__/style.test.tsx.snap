// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme styles renders GlobalStyle correctly 1`] = `null`;

exports[`Theme styles renders Theme correctly 1`] = `
@media (max-width:480px) {

}

<div>
  render div
</div>
`;
